### PR TITLE
STM32Cube packages update

### PR DESCRIPTION
**Pull request by gautierg-st**
New batch of stm32cube package updates:

- update stm32f0 to cube version V1.11.4
- update stm32f3 to cube version V1.11.4
- update stm32l0 to cube version V1.12.2
- update stm32u5 to cube version V1.2.0
- update stm32wb to cube version V1.16.0

### CODE REVIEW

File: stm32cube/stm32f3xx/drivers/include/Legacy/stm32f3xx_hal_can_legacy.h [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
+  * Copyright (c) 2016 STMicroelectronics.
+  * All rights reserved.
   *
-  * This software component is licensed by ST under BSD 3-Clause license,
-  * the "License"; You may not use this file except in compliance with the
-  * License. You may obtain a copy of the License at:
-  *                        opensource.org/licenses/BSD-3-Clause
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
```
Comment (erwango) [inline stm32cube/stm32f3xx/drivers/include/Legacy/stm32f3xx_hal_can_legacy.h:14]: Ditto

File: stm32cube/stm32l0xx/drivers/include/stm32l0xx_hal.h [hunk 2]
```diff
@@ -7,13 +7,12 @@
...
+  * Copyright (c) 2016 STMicroelectronics.
+  * All rights reserved.
   *
-  * This software component is licensed by ST under BSD 3-Clause license,
-  * the "License"; You may not use this file except in compliance with the
-  * License. You may obtain a copy of the License at:
-  *                        opensource.org/licenses/BSD-3-Clause
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
```
Comment (erwango) [inline stm32cube/stm32l0xx/drivers/include/stm32l0xx_hal.h:15]: Ditto

File: lib/stm32wb/hci/README [hunk 3]
```diff
@@ -60,7 +60,7 @@ License Link:
    opensource.org/licenses/BSD-3-Clause
 
 Patch List:
-
+--> please check that the following list
```
Comment (erwango) [inline lib/stm32wb/hci/README:63]: Remove this line

**Review by erwango — CHANGES_REQUESTED** [review]

File: stm32cube/stm32f0xx/drivers/include/Legacy/stm32f0xx_hal_can_legacy.h [hunk 4]
```diff
@@ -6,13 +6,12 @@
...
+  * Copyright (c) 2016 STMicroelectronics.
+  * All rights reserved.
   *
-  * This software component is licensed by ST under BSD 3-Clause license,
-  * the "License"; You may not use this file except in compliance with the
-  * License. You may obtain a copy of the License at:
-  *                        opensource.org/licenses/BSD-3-Clause
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
```
[the remark this answers is not in the document]
Comment (gautierg-st) [reply stm32cube/stm32f0xx/drivers/include/Legacy/stm32f0xx_hal_can_legacy.h:14]: Done

File: lib/stm32wb/hci/README [hunk 5]
```diff
@@ -86,5 +86,19 @@ Patch List:
             used as, in Zephyr context, it is  running on C-M4 side.
           Impacted file: app_conf.h
 
-
-	* Remove trailing whitespaces
+         *  Change payload size of several structure defined in tl.h from 255
```
[the remark this answers is not in the document]
Comment (gautierg-st) [reply lib/stm32wb/hci/README]: Done

File: lib/stm32wb/hci/README (hunk 3 above)
Comment (gautierg-st) [reply lib/stm32wb/hci/README]: Done

File: lib/stm32wb/hci/app_conf.h [hunk 6]
```diff
@@ -278,12 +295,30 @@
...
+ * - SHCI_C2_BLE_INIT_OPTIONS_ENHANCED_ATT_NOTSUPPORTED
+ * which are used to set following configuration bits:
+ * (bit 0): 1: appearance Writable
+ *          0: appearance Read-Only
+ * (bit 1): 1: Enhanced ATT supported
+ *          0: Enhanced ATT not supported
  * other bits: reserved (shall be set to 0)
  */
-/* Z-WB-3 */
-#define CFG_BLE_OPTIONS  (SHCI_C2_BLE_INIT_OPTIONS_LL_ONLY | SHCI_C2_BLE_INIT_OPTIONS_WITH_SVC_CHANGE_DESC | SHCI_C2_BLE_INIT_OPTIONS_DEVICE_NAME_RW | SHCI_C2_BLE_INIT_OPTIONS_NO_EXT_ADV | SHCI_C2_BLE_INIT_OPTIONS_NO_CS_ALGO2 | SHCI_C2_BLE_INIT_OPTIONS_POWER_CLASS_2_3)
```
Comment (erwango) [inline lib/stm32wb/hci/app_conf.h:286]: /* Z-WB-3 */ deleted

**Review by erwango — CHANGES_REQUESTED** [review]

File: lib/stm32wb/hci/app_conf.h (hunk 6 above)
Comment (gautierg-st) [reply lib/stm32wb/hci/app_conf.h]: Fixed

**Review by erwango — APPROVED** [review]

File: lib/stm32wb/hci/app_conf.h [hunk 7]
```diff
@@ -228,12 +240,13 @@
...
- * - bit 1:   1: STM32W5M Module device                              0: Other devices as STM32WBxx SOC, STM32WB1M module
+ * - bit 1:   1: STM32WB5M Module device                             0: Other devices as STM32WBxx SOC, STM32WB1M module
+ * - bit 2:   1: HSE/1024 Clock config                               0: LSE Clock config
  */
 #if defined(STM32WB5Mxx)
-  #define CFG_BLE_LSE_SOURCE  (SHCI_C2_BLE_INIT_CFG_BLE_LSE_NOCALIB | SHCI_C2_BLE_INIT_CFG_BLE_LSE_MOD5MM_DEV)
+  #define CFG_BLE_LS_SOURCE  (SHCI_C2_BLE_INIT_CFG_BLE_LS_NOCALIB | SHCI_C2_BLE_INIT_CFG_BLE_LS_MOD5MM_DEV | SHCI_C2_BLE_INIT_CFG_BLE_LS_CLK_LSE)
 #else
-  #define CFG_BLE_LSE_SOURCE  (SHCI_C2_BLE_INIT_CFG_BLE_LSE_NOCALIB | SHCI_C2_BLE_INIT_CFG_BLE_LSE_OTHER_DEV)
+  #define CFG_BLE_LS_SOURCE  (SHCI_C2_BLE_INIT_CFG_BLE_LS_NOCALIB | SHCI_C2_BLE_INIT_CFG_BLE_LS_OTHER_DEV | SHCI_C2_BLE_INIT_CFG_BLE_LS_CLK_LSE)
```
[the remark this answers is not in the document]
Comment (gautierg-st) [reply lib/stm32wb/hci/app_conf.h:249]: Updated in PR https://github.com/zephyrproject-rtos/zephyr/pull/56530

**Pull request merged**